src/classes/CRLP_Query_SEL.cls

```diff
@@ -56,6 +56,11 @@
             allFieldsToQuery.add(SObjectType.Partial_Soft_Credit__c.fields.Contact__c.getSobjectField().getDescribe());
             allFieldsToQuery.add(SObjectType.Partial_Soft_Credit__c.fields.Amount__c.getSobjectField().getDescribe());
             allFieldsToQuery.add(SObjectType.Partial_Soft_Credit__c.fields.Role_Name__c.getSobjectField().getDescribe());
+        } else if (resultObject == Account_Soft_Credit__c.SObjectType) {
+            allFieldsToQuery.add(SObjectType.Account_Soft_Credit__c.fields.Opportunity__c.getSobjectField().getDescribe());
+            allFieldsToQuery.add(SObjectType.Account_Soft_Credit__c.fields.Account__c.getSobjectField().getDescribe());
+            allFieldsToQuery.add(SObjectType.Account_Soft_Credit__c.fields.Amount__c.getSobjectField().getDescribe());
```

Comment:
Is there a need to force a query of *all* the fields on the object? Generally the ones above are for fields that might be used for special conditions within the code, but where the fields themselves might not be referenced by a rollup or filter rule. In the cases of these 4 fields, I would expect that fields like Amount and Role would always be referenced by a filter rule or rollup. Opportunity and Account though would be needed since those will likely not be referenced by a rule or rollup.

Reply:
I was following the pattern from PSC. Is there a reason it needs Amount and Role always queried that doesn't apply to AccSC? Or should we remove Amount and Role from both objects?

Reply:
@bethbrains  Sorry. I never responded to your question here. Amount and Role on the PSC object are manually added because I believe they can be referenced in one of the query builder utility methods or loops even if the fields are technically included in an active rollups. Basically these two fields really should be there for any soft credit rollup. 

I can see your point of force adding them to the AccountSoftCredit query. It's roughly the same concept, so let's leave them as it. The logic won't hurt anything even if those fields are included dynamically as well.